src/pages/TransactionDetails/components/EventsList/EventItem.tsx

```diff
@@ -0,0 +1,145 @@
+import { useRef, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import {
+  CopyButton,
+  DataDecode,
+  DecodeMethodEnum,
+  AccountLink,
+  NetworkLink
+} from 'components';
+import { urlBuilder } from 'helpers';
+import {
+  useActiveRoute,
+  useGetTransactionUrlHashParams,
+  useScrollToTransactionSection
+} from 'hooks';
+import { faExchange, faSearch } from 'icons/regular';
+import { transactionsRoutes } from 'routes';
+import { EventType } from 'types';
+
+import { EventExtraData } from './EventExtraData';
+
+interface EventItemUIType {
+  event: EventType;
+  txHash: string;
+  id: string;
+}
+
+export const EventItem = ({ event, txHash, id }: EventItemUIType) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const activeRoute = useActiveRoute();
+  const {
+    hashId,
+    hashIndex,
+    hashDecodeMethod,
+    secondHashDecodeMethod,
+    thirdHashDecodeMethod
+  } = useGetTransactionUrlHashParams();
+
+  const isEventHighlighted =
+    hashId === id &&
+    event.order === Number(hashIndex) &&
+    activeRoute(transactionsRoutes.transactionDetailsLogs);
+
+  const [topicsDecodeMethod, setTopicsDecodeMethod] =
+    useState<DecodeMethodEnum>(
+      isEventHighlighted ? hashDecodeMethod : DecodeMethodEnum.raw
+    );
+  const [dataDecodeMethod, setDataDecodeMethod] = useState<DecodeMethodEnum>(
+    isEventHighlighted ? secondHashDecodeMethod : DecodeMethodEnum.raw
+  );
+  const [additionalDataDecodeMethod, setAdditionalDataDecodeMethod] =
+    useState<DecodeMethodEnum>(
+      isEventHighlighted ? thirdHashDecodeMethod : DecodeMethodEnum.raw
+    );
+
+  const dataBase64Buffer = Buffer.from(String(event?.data), 'base64');
+  const dataHexValue = dataBase64Buffer.toString('hex');
+
+  const eventLink = `${urlBuilder.transactionDetailsLogs(txHash)}#${id}/${
+    event.order
+  }/${topicsDecodeMethod}/${dataDecodeMethod}/${additionalDataDecodeMethod}`;
+
+  useScrollToTransactionSection(ref);
+
+  if (!event) {
+    return null;
+  }
+
+  return (
+    <div
+      className={`detailed-item d-flex border-start border-bottom ms-3 py-3 ${
```

Comment:
This could use `classNames` and you would not need to use backticks anymore.

Reply:
updated

Reply:
Great. 🔥